### PR TITLE
Better visualization for tag merge (with itself).

### PR DESCRIPTION
**Pull request by galegosimpatico**
Flash 400 instead of 500 for the invalid request.

Attempts to tackle issue #175.

### CODE REVIEW

File: app/controllers/tags_controller.rb [hunk 1]
```diff
@@ -94,6 +94,26 @@ def select_merge; end
 
   def merge
     @primary = @tag
+
+    # No merge to self.
+    if params[:merge_with_id].eql? @primary.id.to_s
+      flash[:danger] = 'Cannot merge a tag with itself.'
+
+      # redirect_to tag_path(id: @category.id, tag_id: @primary.id) && return
```
Comment (galegosimpatico) [inline app/controllers/tags_controller.rb:102]: This didn't do what I expected, at all.

File: app/controllers/tags_controller.rb [hunk 2]
```diff
@@ -94,6 +94,26 @@ def select_merge; end
...
+
+      # render :edit, status: 400
```
Comment (galegosimpatico) [inline app/controllers/tags_controller.rb:104]: This worked better than the previous one, but didn't do what anyone would expect it the page to do, TBH I was seeing it coming.

File: app/controllers/tags_controller.rb [hunk 3]
```diff
@@ -94,6 +94,26 @@ def select_merge; end
...
+      # return
+
+      # render json: { success: false, status: 400, tag: @tag }
```
Comment (galegosimpatico) [inline app/controllers/tags_controller.rb:107]: This was a very bad idea that seemed not so bad until executed.

File: app/controllers/tags_controller.rb [hunk 4]
```diff
@@ -94,6 +94,26 @@ def select_merge; end
...
+      # return
+
+      # render 'tag', category: @category, tag: @tag
```
Comment (galegosimpatico) [inline app/controllers/tags_controller.rb:110]: This wasn't tested very much, I'm no longer even sure it worked at all.

File: app/controllers/tags_controller.rb [hunk 5]
```diff
@@ -94,6 +94,26 @@ def select_merge; end
...
+      # return
+
+      redirect_back fallback_location: categories_path
```
Comment (galegosimpatico) [inline app/controllers/tags_controller.rb:113]: This didn't what I was expecting it to do either, at all, but I think the results were unexpectedly great.

File: app/controllers/tags_controller.rb (hunk 1 above)
Comment (galegosimpatico) [reply app/controllers/tags_controller.rb:102]: applying technique https://github.com/codidact/qpixel/pull/178/commits/dfae8b9f0a486aca417804fbf14e2d44308841da uses fixes it, but https://github.com/codidact/qpixel/pull/176#discussion_r487302984 is actually better

File: app/controllers/tags_controller.rb [hunk 6]
```diff
@@ -94,6 +94,26 @@ def select_merge; end
 
   def merge
     @primary = @tag
+
+    # No merge to self.
+    if params[:merge_with_id].eql? @primary.id.to_s
```
Comment (ArtOfCode-) [inline app/controllers/tags_controller.rb:99]:
```ruby
if params[:merge_with_id] == @primary.id.to_s
```

**Review by ArtOfCode- — CHANGES_REQUESTED** [review]

File: app/controllers/tags_controller.rb (hunk 6 above)
Comment (galegosimpatico) [reply app/controllers/tags_controller.rb]: I tested it and then committed

**Pull request merged**